emodelrunner/protocols/create_protocols.py

```diff
@@ -0,0 +1,448 @@
+"""Protocol creation functions & custom protocol classes."""
+
+# Copyright 2020-2021 Blue Brain Project / EPFL
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from bluepyopt import ephys
+
+from emodelrunner.create_recordings import get_pairsim_recordings
+from emodelrunner.create_stimuli import load_pulses
+from emodelrunner.configuration import PackageType
+from emodelrunner.protocols import synplas_protocols
+
+from emodelrunner.synapses.recordings import SynapseRecordingCustom
+from emodelrunner.stimuli import MultipleSteps
+from emodelrunner.features import define_efeatures
+from emodelrunner.protocols.reader import ProtocolParser
+from emodelrunner.synapses.create_locations import get_syn_locs
+from emodelrunner.synapses.stimuli import (
+    NrnVecStimStimulusCustom,
+    NetConSpikeDetector,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class ProtocolBuilder:
+    """Class representing the protocols applied in SSCX.
+
+    Attributes:
+        protocols (bluepyopt.ephys.protocols.SequenceProtocol): the protocols to apply to the cell
+    """
+
+    def __init__(self, protocols):
+        """Constructor to be called by the classmethod overloads."""
```

Comment:
I actually added `protocols` as attributes to the class and Sphinx is showing it.

```
class ProtocolBuilder:
    """Class representing the protocols applied in SSCX.

    Attributes:
        protocols (bluepyopt.ephys.protocols.SequenceProtocol): the protocols to apply to the cell
    """

    def __init__(self, protocols):
        """Constructor to be called by the classmethod overloads."""
        self.protocols = protocols
```
![Screenshot 2022-01-24 at 12 26 53](https://user-images.githubusercontent.com/7026020/150774969-3006df73-caad-41a1-ae4c-93f205aca148.png)

Reply:
I added missing return statements here for the classmethods in 10b7b87
I skipped the `_get_syn_locs` method since the private/protected methods are not part of the API and not rendered by Sphinx.

Reply:
I think we should also add protocols as argument of the init fct in the docstring, even if it's redundant with the attributes.

Reply:
Added in f3e43ef